### DISCUSSION TITLE
Problem installing OpenPNM and Porespy in the editable mode

### DISCUSSION THREAD

**Discussion by DMaxJ** [PMEAL/OpenPNM#2375] [Q&A] [answered]
I am back again with a problem related to installation of OpenPNM and Porespy. I had a similar issue before (https://github.com/PMEAL/OpenPNM/issues/1118). 

I am trying to install both in a new computer with Windows 10 in the editable mode.

1. I am using anaconda latest version: Anaconda3-2021.11-Windows-x86_64
2. I cloned the openpnm and porespy repositories from github
3. When I follow the steps mentioned in (https://openpnm.org/user_guide/installation.html), I run in to problems.
4. I added conda forge using `conda config --add channels conda-forge` before following the installation guide. Later I find out that a lot of the requirements are missed. I cannot remember all the details because I tried to many different things to get around this problem.
5. Tried installing inside a virtual environment but ran into problems. OpenPNM was installed, but porespy required visual C++2014 or higher. Tried getting that too, but failed. Moreover, I need to install Spyder within the environment if I want to use Spyder.

Is there any way to solve this problem? Thanks in advance. @jgostick @ma-sadeghi

**Comment by ma-sadeghi** [comment, marked as answer]
Regarding Spyder, you don't need to install Spyder in the same environment. See [here](https://docs.spyder-ide.org/current/faq.html#using-existing-environment) for more details.

**Reply by ma-sadeghi** [reply]
I just gave it a try, installing both `openpnm` and `porespy` from scratch, and it worked. Here's a demo:

https://user-images.githubusercontent.com/14086031/164073456-7122c935-c502-4799-b124-9fca99426d4f.mp4

Note: `mamba` is drop-in replacement for `conda`, but much faster (there's more to it, but that's pretty much it as far as we're concerned).

**Reply by ma-sadeghi** [reply]
This is on Linux though, it might be a Windows issue. Currently, I'm not on Windows, but I'll try to reproduce in Windows when I get the time.

**Reply by DMaxJ** [reply]
Thank you for the solution. Finally I was able to install and run in the editable mode. I did the following

1. Created a virtual environment with python3.9  (probably it did not work when I tried with 3.10)
2. Then installed the requirements with conda for both openpnm and porespy
3. Installed both openpnm and porespy in the editable mode
4. Only problem that I had was when I changed my interpreter option in Spyder, it failed and I had to install Spyder kernel inside my virtual environment.
5. After that I could work with both the tools in Spyder.

Thanks again

**Reply by ma-sadeghi** [reply]
Glad you figured it out @DMaxJ. The reason your Spyder failed when changing interpreter is probably because you don't have `spyder-kernels` installed. You basically need to install `spyder-kernels` in the virtualenv that you plan to use in Spyder. Just do `mamba install spyder-kernels` and it should work just fine.
